### ISSUE TITLE
dev-haskell/hledger-1.22 conflicts

### ISSUE THREAD

**Issue by trofi** [gentoo-haskell/gentoo-haskell#1200]
```
  (dev-haskell/hledger-lib-1.22:0/1.22::haskell, ebuild scheduled for merge) USE="doc hscolour profile test -hoogle" ABI_X86="(64)" pulled in by
    >=dev-haskell/hledger-lib-1.22:= required by (dev-haskell/hledger-ui-1.22:0/0::haskell, ebuild scheduled for merge) USE="threaded" ABI_X86="(64)"
    ^^                        ^^^^                                                                                                                                                                                   
    >=dev-haskell/hledger-lib-1.22:=[profile?] required by (dev-haskell/hledger-1.22:0/1.22::haskell, ebuild scheduled for merge) USE="doc hscolour profile terminfo test threaded -hoogle" ABI_X86="(64)"
    ^^                        ^^^^                                                                                                                                                                                                                                                                                                                                                                                                        
    >=dev-haskell/hledger-lib-1.22:=[profile?] required by (dev-haskell/hledger-web-1.22:0/1.22::haskell, ebuild scheduled for merge) USE="doc hscolour profile test threaded -hoogle -library-only" ABI_X86="(64)"
    ^^                        ^^^^                                                                                                                                                                                                                                                                                                                                                                                                        

  (dev-haskell/hledger-lib-1.21:0/1.21::haskell, installed) USE="doc hscolour profile test -hoogle" ABI_X86="(64)" pulled in by
    <dev-haskell/hledger-lib-1.22:= required by (dev-haskell/hledger-interest-1.6.1:0/0::haskell, ebuild scheduled for merge) USE="" ABI_X86="(64)"
    ^                        ^^^^ ^                                                                                                                                                                                  
    <dev-haskell/hledger-lib-1.22:=[profile?] required by (dev-haskell/hledger-iadd-1.3.14:0/1.3.14::haskell, ebuild scheduled for merge) USE="doc hscolour profile test -hoogle" ABI_X86="(64)"
    ^                        ^^^^ ^       
```

**Comment by solpeth** [comment]
Sorry again, I will sort this.

**Comment by solpeth** [comment]
I've reverted for now. Unfortunately I do not have the time to properly figure out the issues with hledger-interest and hledger-iadd, which won't build due to type signature changes in hledger-lib. I'll try to avoid this situation in the future.

**Comment by trofi** [comment]
Ah, no problem! I can also have a look at how large the API changes are and attempt to port.

**Comment by solpeth** [comment]
No worries!

The main issue is around a change from a list type to a Map in hledger-lib as regards `MixedAmount`, if I remember correctly, as well as hiding some types from the exported API. Importing `Hledger.Data.Types` resolves one issue, the next is adapting functions to the type signature changes in hledger-lib-1.22.

If I didn't have work I'd try to get this done instead of waiting for upstream.

**Comment by solpeth** [comment]
Should be resolved with the release of `hledger-interest-1.6.2` recently.
